include/odp/api/abi-default/atomic.h

```diff
@@ -59,6 +59,31 @@ typedef struct ODP_ALIGNED(sizeof(uint64_t)) odp_atomic_u64_s {
 
 #endif
 
+#ifdef __ARM_FEATURE_ATOMICS
```

Comment:
Both, aarch64 and x86-64 ISAs support 2x64bit CAS operation. So, the decision between locked vs. lock-free implementation cannot be based on an ARM pre-processor macro.

Also, ARM specific macros should not be used on generic files, but under platform/arch/aarch64 files. Generic files may use e.g. GCC macros.

Reply:
__GCC_HAVE_SYNC_COMPARE_AND_SWAP_16 is not set for ARM CASP instructions. Instead of using __ARM_FEATURE_ATOMICS, can we define a ODP macro which can be set/unset by the application ? 

I have moved the ARM specific assembly to platform/linux-generic/arch/aarch64/odp_atomic.h file.